functions.go

```diff
@@ -355,3 +355,15 @@ func Is(err, target error) bool {
 func As(err error, target interface{}) bool {
 	return stderrors.As(err, target)
 }
+
+// SetLocation takes a given error and records where in the stack SetLocation
+// was called from and returns wrapped error with the location information set.
+// The returned error implements the Locationer interface. If err is nil then a
+// nil error is returned.
+func SetLocation(err error, callDepth int) error {
+	if err == nil {
+		return nil
+	}
+
+	return newLocationError(err, callDepth)
```

Comment:
What if err is already a locationErr? ie does it make sense to call SetLocation twice on a given err?

Reply:
Yeah it does. It's the same as calling Trace over and over.